adapters/repos/db/vector/common/sharded_locks.go

```diff
@@ -88,8 +106,21 @@ func NewShardedRWLocks(count uint64) *ShardedRWLocks {
 	}
 
 	return &ShardedRWLocks{
-		shards: make([]sync.RWMutex, count),
-		count:  count,
+		shards:   make([]sync.RWMutex, count),
+		count:    count,
+		PageSize: DefaultPageSize,
+	}
+}
+
+func NewShardedRWLocksWithPageSize(count uint64, pageSize uint64) *ShardedRWLocks {
```

Comment:
I don't see where we ever use count != to 512 and we no longer use `DefaultShardedLocksCount`. Does it make sense to remove the count parameter and just set to `DefaultShardedLocksCount`?